### DISCUSSION TITLE
Weird rendering issues when resizing with custom clip path

### DISCUSSION THREAD

**Discussion by DaanDeReus** [fabricjs/fabric.js#10359] [Q&A]
Hello, i made a custom renderer for border radius. (like figma) the normal behavior is weird in my opinion. That it becomes some stretched corner or whatever.
So i made a clippath Path() that has custom parameters. Now it has to re-render the clippath while and after resizing to ensure it stays not stretched.

It might just be me, i have a niece case.
I can't disclose too much of the code..
This is the most relevant part:

```
canvas.on('object:scaling', (e) => {
   const target = e.target;
   console.log(target);
   
   if (target.borderRadius) {
      const newWidth = target.width * target.scaleX;
      const newHeight = target.height * target.scaleY;

      const maxRadius = Math.min(newWidth / 2, newHeight / 2);
      const currentRadius = Math.min(target.borderRadius, maxRadius);

      target.set({
         scaleX: 1,
         scaleY: 1,
         width: newWidth,
         height: newHeight,
         clipPath: generateClipPath(newWidth, newHeight, redacted)
      });
   }
});
```
generateClipPath returns a Path().

I have a control that can edit the size of the border radius. But that is not relevant.

here is a video of the issue:

https://github.com/user-attachments/assets/323f81ce-6bf6-4a70-a926-073038918972

How the object is rendered:

```
const rect = new Rect({
   left: 100,
   top: 100,
   width: 200,
   height: 100,
   fill: 'red',
   objectCaching: true,
});
rect.controls = {
   ...rect.controls,
   borderRadiusControl,
};
canvas.add(rect);
```

Thanks in advance!

**Comment by asturur** [comment]
The normal behaviour is weird because you want to treat scaleX and scaleY as a width and height change.
By not just writing your custom control you are putting yourself in a much more complicated situation.

**Reply by DaanDeReus** [reply]
Yeah i know. Its just that the clippath thing doesn't work anymore if i dont do that. I will just keep trying then. Thanks
